network/network.go

```
@@ -5,40 +5,46 @@ import (
 	"github.com/apache/thrift/lib/go/thrift"
 )
 
-func NewAgentServer(handler env.Agent, addr string) *thrift.TSimpleServer {
-	transportFactory := thrift.NewTBufferedTransportFactory(8192)
+func NewAgentServer(handler env.Agent, addr string) (*thrift.TSimpleServer, error) {
 	transport, _ := thrift.NewTServerSocket(addr)
+	if err := transport.Open(); err != nil {
+		return nil, err
+	}
 	processor := env.NewAgentProcessor(handler)
-	protocolFactory := thrift.NewTBinaryProtocolFactoryConf(nil)
-	return thrift.NewTSimpleServer4(processor, transport, transportFactory, protocolFactory)
+	return thrift.NewTSimpleServer2(processor, transport), nil
 }
 
-func NewWorldServer(handler env.World, addr string) *thrift.TSimpleServer {
-	transportFactory := thrift.NewTBufferedTransportFactory(8192)
+func NewWorldServer(handler env.World, addr string) (*thrift.TSimpleServer, error) {
 	transport, _ := thrift.NewTServerSocket(addr)
+	if err := transport.Open(); err != nil {
+		return nil, err
+	}
 	processor := env.NewWorldProcessor(handler)
-	protocolFactory := thrift.NewTBinaryProtocolFactoryConf(nil)
-	return thrift.NewTSimpleServer4(processor, transport, transportFactory, protocolFactory)
+	return thrift.NewTSimpleServer2(processor, transport), nil
 }
 
-func NewAgentClient(addr string) *env.AgentClient {
+func NewAgentClient(addr string) (*env.AgentClient, error) {
 	transportFactory := thrift.NewTBufferedTransportFactory(8192)
 	transportSocket := thrift.NewTSocketConf(addr, nil)
 	transport, _ := transportFactory.GetTransport(transportSocket)
```

don't ignore the error here, since we can now surface it easily to the caller